src/alg/dpll.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+
+#######################################################################################################
+########### FUNCTIONS USED IN DPLL ####################################################################
+#######################################################################################################
+
+# simplify the cnf by assigning one of the variable to true or false and deleating clauses and literals 
+def conditioning(cnf,var):
+    new_cnf = list()
+    for entry in cnf:
+        if (len(entry)==0):
+            new_cnf.append(np.array([]))
+        elif var not in entry:
+            if -var in entry:
+                index = np.argwhere(entry==-var)
+                new_entry = np.delete(entry,index)
+                new_cnf.append(new_entry)
+            else:
+                new_cnf.append(entry)  
+    return new_cnf
+
+# find the most variables occouring most often across al clauses 
+def findMostCommonVar(cnf):
+    # create an array containing all the variables as often as they occour in the clauses (in either negated and non-negated state)
+    knf_variables_list = np.array([])
+    for array in cnf:
+        for entry in array:
+            knf_variables_list = np.append(knf_variables_list,abs(entry))
```

Comment:
I'm not sure, if this is unnecessary pushing the runtime, of the DPLL or if the specific choice is compensating for this, but we should take care of this on the long run, i think

Reply:
I do think that we will probably use more advanced methods of choosing the next variable to try in the long run, I see this rather as a starting point as well and figured this would probably be better than choosing a variable completely at random! :)